app/containers/RoomItem/index.tsx

```diff
@@ -23,149 +26,106 @@ const attrs = [
-		if (prevProps.connected !== connected && connected && this.isDirect) {
-			getUserPresence(id);
-		}
-	}
-
-	componentWillUnmount() {
-		if (this.roomSubscription?.unsubscribe) {
-			this.roomSubscription.unsubscribe();
-		}
-	}
-
-	get isGroupChat() {
-		const { item, getIsGroupChat } = this.props;
-		return getIsGroupChat(item);
-	}
-
-	get isDirect() {
-		const {
-			item: { t },
-			id
-		} = this.props;
-		return t === 'd' && id && !this.isGroupChat;
-	}
-
-	init = () => {
-		const { item } = this.props;
-		if (item?.observe) {
-			const observable = item.observe();
-			this.roomSubscription = observable?.subscribe?.(() => {
-				this.forceUpdate();
-			});
-		}
-	};
-
-	onPress = () => {
-		const { item, onPress } = this.props;
-		return onPress(item);
-	};
-
-	onLongPress = () => {
-		const { item, onLongPress } = this.props;
-		if (onLongPress) {
-			return onLongPress(item);
-		}
-	};
-
-	render() {
-		const {
-			item,
-			getRoomTitle,
-			getRoomAvatar,
-			getIsRead,
-			width,
-			toggleFav,
-			toggleRead,
-			hideChannel,
-			theme,
-			isFocused,
-			status,
-			showLastMessage,
-			username,
-			useRealName,
-			swipeEnabled,
-			autoJoin,
-			showAvatar,
-			displayMode
-		} = this.props;
+let roomSubscription: Subscription | null;
+let accessibilityLabel = '';
+
+const RoomItemContainer = React.memo(
+	({
+		item,
+		id,
+		onPress,
+		onLongPress,
+		width,
+		toggleFav,
+		toggleRead,
+		hideChannel,
+		isFocused,
+		showLastMessage,
+		username,
+		useRealName,
+		autoJoin,
+		showAvatar,
+		displayMode,
+		getRoomTitle = () => 'title',
+		getRoomAvatar = () => '',
+		getIsRead = () => false,
+		swipeEnabled = true
+	}: IRoomItemContainerProps) => {
 		const name = getRoomTitle(item);
 		const testID = `rooms-list-view-item-${name}`;
 		const avatar = getRoomAvatar(item);
 		const isRead = getIsRead(item);
 		const date = item.roomUpdatedAt && formatDate(item.roomUpdatedAt);
 		const alert = item.alert || item.tunread?.length;
-
-		let accessibilityLabel = name;
-		if (item.unread === 1) {
-			accessibilityLabel += `, ${item.unread} ${I18n.t('alert')}`;
-		} else if (item.unread > 1) {
-			accessibilityLabel += `, ${item.unread} ${I18n.t('alerts')}`;
-		}
-
-		if (item.userMentions > 0) {
-			accessibilityLabel += `, ${I18n.t('you_were_mentioned')}`;
-		}
-
-		if (date) {
-			accessibilityLabel += `, ${I18n.t('last_message')} ${date}`;
-		}
+		const connected = useAppSelector(state => state.meteor.connected);
+		const userStatus = useAppSelector(state => state.activeUsers[id || '']?.status);
+		const [_, forceUpdate] = useReducer(x => x + 1, 0);
+		const { theme } = useTheme();
+
+		useEffect(() => {
+			const init = () => {
+				if (item?.observe) {
+					const observable = item.observe();
+					roomSubscription = observable?.subscribe?.(() => {
+						if (_) forceUpdate();
+					});
+				}
+				return () => roomSubscription?.unsubscribe && roomSubscription.unsubscribe();
```

Comment:
Shorter without hurting readability 🙈 

```suggestion
				return () => roomSubscription?.unsubscribe?.();
```